utils/compare_ds.py

```diff
@@ -95,6 +96,42 @@ def print_offending_elements(elements, sign):
         print("%s %s %s" % (sign, thing, atrribute))
 
 
+def compare_platforms(old_rule, new_rule, old_benchmark, new_benchmark):
+    entries = [{
+            "benchmark": old_benchmark,
+            "rule": old_rule,
+            "cpe": []
+        }, {
+            "benchmark": new_benchmark,
+            "rule": new_rule,
+            "cpe": []
+        }]
+
+    for entry in entries:
+        for platform in entry["rule"].findall(".//xccdf:platform", ns):
+            idref = platform.get("idref")
+            if idref.startswith("#"):
+                cpe_platforms = entry["benchmark"].findall(
+                    ".//cpe-lang:platform[@id='{0}']".format(idref.replace("#", "")), ns)
+                if len(cpe_platforms) > 1:
+                    print("Platform {0} defined more than once".format(idref))
+                    for cpe_p in cpe_platforms:
+                        ET.dump(cpe_p)
+                if len(cpe_platforms) == 0:
+                    print("Platform {0} not defined in platform specification".format(idref))
+                    break
+                for cpe_platform in cpe_platforms:
+                    fact_refs = cpe_platform.findall(".//cpe-lang:fact-ref", ns)
+                    for fact_ref in fact_refs:
+                        entry["cpe"].append(fact_ref.get("name"))
+            else:
+                entry["cpe"].append(idref)
+
+    if entries[0]["cpe"] != entries[1]["cpe"]:
+        print("Platform has been changed for rule '{0}': {1} -> {2}".format(
+            entries[0]["rule"].get("id"), repr(entries[0]["cpe"]), repr(entries[1]["cpe"])))
```

Comment:
Let's make the output more like the other diffs

something like:

```suggestion
        print("Platform definition %s differs for:" % (entries[0]["rule"].get("id")))
        print("--- old datastream")
        print("+++ new datastream")
        print("-{}".format(repr(entries[0]["cpe"])))
        print("+{}".format(repr(entries[1]["cpe"])))
```

This way we have nice colors when displaying the diff... even though the old and new entries are not really code, it will popup for user's eyes

Reply:
Sure, done